demes/src/specification.rs

```diff
@@ -1059,6 +1059,49 @@ impl Deme {
     pub fn proportions(&self) -> &[Proportion] {
         &self.proportions
     }
+
+    /// Size of Deme at a given time
+    pub fn size_at<F: Into<f64>>(&self, time: F) -> Result<DemeSize, DemesError> {
+        let time: f64 = time.into();
+
+        if time == f64::INFINITY && self.start_time == f64::INFINITY {
+            return Ok(self.epochs()[0].start_size);
+        }
+
+        let epochs: Vec<Epoch> = self
+            .epochs()
+            .iter()
+            .filter(|x| {
+                x.time_interval()
+                    .contains_inclusive_start_exclusive_end(time)
+            })
+            .cloned()
+            .collect();
+
+        let epoch = match epochs.len() {
+            1 => epochs[0],
+            _ => {
+                return Err(DemesError::EpochError(
+                    "No Epoch corresponding to this time".to_string(),
```

Comment:
I am not sure that "no epochs" is an error.  I can imagine code iterating over all deme ids and asking "get me all the sizes for time X".  When no epoch exists, we could return either None (meaning the return type is Result<Option<DemeSize>, DemesError>> or 0.0.

Reply:
Related: changing to returning Option<Time> means no error path appears possible at the moment.  But, because deme sizes are f64, "bad things" can happen and we want to make sure that results are positive and finite before returning, implying that there are actual error paths that need handling (even if they are hard to trigger).

Reply:
That makes sense.
I'll try to switch to Option and validate the DemeSize before returning.

Reply:
I've switched to using an Option.
The thing I am unsure about is the DemesError enum case I should use, right now it's DemeError. I can change that to another one.

Reply:
Yeah, that's a good question. One could argue two ways: we should have a LibraryError enum variant to handle "bad stuff happening internally" OR this happened when asking about a Deme, so it is a DemeError.  I am happy either way.

Reply:
We don't need to decide this now: this is a new feature and the next stable release isn't imminent.  So we have time.